cmd/agent/main.go

```diff
@@ -117,15 +129,17 @@ func GetMetrics(chanSend chan int64, chanMetrics chan []data.Metrics, timer time
 			} else {
 				pollCount = signal
 			}
+		case <-ctx.Done():
+			return
 		default:
 			time.Sleep(timer * time.Second)
 		}
 		pollCount += 1
 	}
 }
 
-// GetMetricsUtil - function, that collects total/free memory and utilizatin of every cpu. 
-func GetMetricsUtil(chanSend chan int64, chanMetrics chan []data.Metrics, timer time.Duration) {
+// GetMetricsUtil - function, that collects total/free memory and utilizatin of every cpu.
+func GetMetricsUtil(chanSend chan int64, chanMetrics chan []data.Metrics, timer time.Duration, ctx context.Context) {
```

Comment:
```suggestion
func GetMetricsUtil(ctx context.Context, chanSend chan int64, chanMetrics chan []data.Metrics, timer time.Duration) {
```
В Go принято контекст всегда передавать первым параметром.